id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/controller/IdRepoDraftController.java

```diff
@@ -0,0 +1,171 @@
+package io.mosip.idrepository.identity.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.validation.Errors;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.WebDataBinder;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.InitBinder;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.mosip.idrepository.core.constant.AuditEvents;
+import io.mosip.idrepository.core.constant.AuditModules;
+import io.mosip.idrepository.core.constant.IdRepoErrorConstants;
+import io.mosip.idrepository.core.constant.IdType;
+import io.mosip.idrepository.core.dto.IdRequestDTO;
+import io.mosip.idrepository.core.dto.IdResponseDTO;
+import io.mosip.idrepository.core.exception.IdRepoAppException;
+import io.mosip.idrepository.core.helper.AuditHelper;
+import io.mosip.idrepository.core.logger.IdRepoLogger;
+import io.mosip.idrepository.core.security.IdRepoSecurityManager;
+import io.mosip.idrepository.core.spi.IdRepoDraftService;
+import io.mosip.idrepository.core.util.DataValidationUtil;
+import io.mosip.idrepository.identity.validator.IdRequestValidator;
+import io.mosip.kernel.core.logger.spi.Logger;
+import springfox.documentation.annotations.ApiIgnore;
+
+/**
+ * @author Manoj SP
+ *
+ */
+@RestController
+public class IdRepoDraftController {
+
+	private static final String ID_REPO_DRAFT_CONTROLLER = "IdRepoDraftController";
+
+	private final Logger mosipLogger = IdRepoLogger.getLogger(IdRepoDraftController.class);
+
+	@Autowired
+	private IdRequestValidator validator;
+
+	@Autowired
+	private IdRepoDraftService<IdRequestDTO, IdResponseDTO> draftService;
+
+	@Autowired
+	private AuditHelper auditHelper;
+
+	@InitBinder
+	public void initBinder(WebDataBinder binder) {
+		binder.addValidators(validator);
+	}
+
+	@PreAuthorize("hasAnyRole('REGISTRATION_PROCESSOR')")
+	@PostMapping(path = "/createDraft", consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<IdResponseDTO> createDraft(@Validated @RequestBody IdRequestDTO request, @ApiIgnore Errors errors)
```

Comment:
The RegId can be a path variable in create and update methods also.

Reply:
@ramesh-n I have updated